sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2397,7 +2397,11 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
                 s"contains illegal character for hexBinary: $padding$value");
           }
         case other =>
-          throw QueryParsingErrors.literalValueTypeUnsupportedError(other, ctx)
+          throw QueryParsingErrors.literalValueTypeUnsupportedError(
+            unsupportedType = other,
+            supportedTypes =
+              Seq("DATE", "TIMESTAMP_NTZ", "TIMESTAMP_LTZ", "TIMESTAMP", "INTERVAL", "X"),
```

Comment:
What's X? Hexadecimal?

Reply:
Yep

Reply:
Maybe a lowercase `hexadecimal` would be less confusing?

Reply:
Here, I pointed out the exact keyword that can be used in the typed literals, see
https://github.com/apache/spark/blob/933dc0c42f0caf74aaa077fd4f2c2e7208452b9b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L2390

Reply:
We could give a hint to user like `X hexadecimal`

Reply:
Do you mean `"X"` -> `"X hexadecimal"` ?

Reply:
Agree with you, `"X hexadecimal"` looks a little strange